The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6436540665/job/17480077833) [(plain)](https://github.com/rust-lang/rust/commit/08f10c22b3d20b66126447046718744622e82428/checks/17480077833/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=eswartz
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_241d8237-72b0-480d-b220-e675d7a85f62
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=docs/total_cmp-test-result-in-docs
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_241d8237-72b0-480d-b220-e675d7a85f62
GITHUB_REF=refs/pull/115412/merge
GITHUB_REF_NAME=115412/merge
GITHUB_REF_PROTECTED=false
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=0465d59e1297cb2c77418e50767e6c5cb5729233
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_241d8237-72b0-480d-b220-e675d7a85f62
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_241d8237-72b0-480d-b220-e675d7a85f62
GITHUB_TRIGGERING_ACTOR=eswartz
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/115412/merge
GITHUB_WORKFLOW_SHA=0465d59e1297cb2c77418e50767e6c5cb5729233
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
..i....................................................................................

failures:

---- library/core/src/num/f32.rs - f32::f32::total_cmp (line 1430) stdout ----
error[E0425]: cannot find value `bois` in this scope
error: doctest failed, to rerun pass `-p core --doc`
  |
  |
4 | assert!(bois.into_iter().map(|b| b.weight)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
For more information about this error, try `rustc --explain E0425`.
Couldn't compile the test.
---- library/core/src/num/f32.rs - f32::f32::total_cmp (line 1436) stdout ----
error[E0425]: cannot find value `bois` in this scope
  |
  |
4 | assert!(bois.into_iter().map(|b| b.weight)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
For more information about this error, try `rustc --explain E0425`.
Couldn't compile the test.
---- library/core/src/num/f64.rs - f64::f64::total_cmp (line 1428) stdout ----
error[E0425]: cannot find value `bois` in this scope
  |
  |
4 | assert!(bois.into_iter().map(|b| b.weight)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
For more information about this error, try `rustc --explain E0425`.
Couldn't compile the test.
---- library/core/src/num/f64.rs - f64::f64::total_cmp (line 1434) stdout ----
error[E0425]: cannot find value `bois` in this scope
  |
  |
4 | assert!(bois.into_iter().map(|b| b.weight)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
```

</details>

--- a/library/core/src/num/f32.rs
+++ b/library/core/src/num/f32.rs
@@ -1424,9 +1424,17 @@ impl f32 {
     /// ];
     ///
     /// bois.sort_by(|a, b| a.weight.total_cmp(&b.weight));
-    /// # assert!(bois.into_iter().map(|b| b.weight)
-    /// #     .zip([-5.0, 0.1, 10.0, 99.0, f32::INFINITY, f32::NAN].iter())
-    /// #     .all(|(a, b)| a.to_bits() == b.to_bits()))
+    ///
+    /// // `f32::NAN` could be positive or negative, which will affect the sort order.
+    /// if f32::NAN.is_sign_negative() {
+    ///     assert!(bois.into_iter().map(|b| b.weight)
+    ///         .zip([f32::NAN, -5.0, 0.1, 10.0, 99.0, f32::INFINITY].iter())
+    ///         .all(|(a, b)| a.to_bits() == b.to_bits()))
+    /// } else {
+    ///     assert!(bois.into_iter().map(|b| b.weight)
+    ///         .zip([-5.0, 0.1, 10.0, 99.0, f32::INFINITY, f32::NAN].iter())
+    ///         .all(|(a, b)| a.to_bits() == b.to_bits()))
+    /// }
     /// ```
     #[stable(feature = "total_cmp", since = "1.62.0")]
     #[must_use]

--- a/library/core/src/num/f64.rs
+++ b/library/core/src/num/f64.rs
@@ -1422,9 +1422,17 @@ impl f64 {
     /// ];
     ///
     /// bois.sort_by(|a, b| a.weight.total_cmp(&b.weight));
-    /// # assert!(bois.into_iter().map(|b| b.weight)
-    /// #     .zip([-5.0, 0.1, 10.0, 99.0, f64::INFINITY, f64::NAN].iter())
-    /// #     .all(|(a, b)| a.to_bits() == b.to_bits()))
+    ///
+    /// // `f64::NAN` could be positive or negative, which will affect the sort order.
+    /// if f64::NAN.is_sign_negative() {
+    ///     assert!(bois.into_iter().map(|b| b.weight)
+    ///         .zip([f64::NAN, -5.0, 0.1, 10.0, 99.0, f64::INFINITY].iter())
+    ///         .all(|(a, b)| a.to_bits() == b.to_bits()))
+    /// } else {
+    ///     assert!(bois.into_iter().map(|b| b.weight)
+    ///         .zip([-5.0, 0.1, 10.0, 99.0, f64::INFINITY, f64::NAN].iter())
+    ///         .all(|(a, b)| a.to_bits() == b.to_bits()))
+    /// }
     /// ```
     #[stable(feature = "total_cmp", since = "1.62.0")]
     #[must_use]

